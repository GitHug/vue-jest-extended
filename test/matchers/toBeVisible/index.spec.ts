import FakeWrapper from '../../utils';
import matcher from '../../../src/matchers/toBeVisible';

expect.extend(matcher);

describe('.toBeVisible', () => {
  const wrapper = new FakeWrapper();

  test('passes when wrapper is visible', () => {
    wrapper.visible = true;
    expect(wrapper).toBeVisible();
  });

  test('fails when wrapper is not visible', () => {
    wrapper.visible = false;
    expect(() => {
      expect(wrapper).toBeVisible();
    }).toThrowErrorMatchingSnapshot();
  });

  describe('.not.toBeVisible', () => {
    test('passes when wrapper does not exist', () => {
      wrapper.visible = false;
      expect(wrapper).not.toBeVisible();
    });

    test('fails when wrapper exists', () => {
      wrapper.visible = true;
      expect(() => {
        expect(wrapper).not.toBeVisible();
      }).toThrowErrorMatchingSnapshot();
    });
  });
});
