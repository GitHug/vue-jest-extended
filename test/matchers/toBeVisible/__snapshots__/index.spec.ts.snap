// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.toBeVisible .not.toBeVisible fails when wrapper exists 1`] = `
"
  [2mexpect([22m[31mreceived[39m[2m).not.toBeVisible([22m[32mexpected[39m[2m)[22m


  Expected wrapper not to be visible:

    [32mfalse[39m

  received:
    [31mtrue[39m
  "
`;

exports[`.toBeVisible fails when wrapper is not visible 1`] = `
"
  [2mexpect([22m[31mreceived[39m[2m).toBeVisible([22m[32mexpected[39m[2m)[22m


  Expected wrapper to be visible:

    [32mtrue[39m

  received:
    [31mfalse[39m
  "
`;
